src/main/java/com/coveros/selenified/Selenified.java

```diff
@@ -342,28 +341,30 @@ protected void startTest(Object[] dataProvider, Method method, ITestContext test
         }
         int invocationCount = (int) test.getAttribute(testName + INVOCATION_COUNT);
 
-        Browser myBrowser = browsers.get(invocationCount);
+        Capabilities capabilities = Selenified.capabilities.get(invocationCount);
+        capabilities.setInstance(invocationCount);
+        Browser browser = capabilities.getBrowser();
         if (!selenium.useBrowser()) {
-            myBrowser = new Browser(BrowserName.NONE);
+            browser = new Browser("None");
         }
-        DesiredCapabilities myCapability = capabilities.get(invocationCount);
-        myCapability.setCapability("name", testName);
-        this.capability.set(myCapability);
+        DesiredCapabilities desiredCapabilities = capabilities.getDesiredCapabilities();
+        desiredCapabilities.setCapability("name", testName);
+        this.capability.set(desiredCapabilities);
 
         OutputFile outputFile =
-                new OutputFile(outputDir, testName, myBrowser, getTestSite(extClass, test), test.getName(), group,
+                new OutputFile(outputDir, testName, capabilities, getTestSite(extClass, test), test.getName(), group,
                         getAuthor(extClass, test), getVersion(extClass, test), description);
         if (selenium.useBrowser()) {
             App app = null;
             try {
-                app = new App(myBrowser, myCapability, outputFile);
+                app = new App(capabilities, outputFile);
             } catch (InvalidBrowserException | MalformedURLException e) {
                 log.error(e);
             }
             this.apps.set(app);
             this.calls.set(null);
             outputFile.setApp(app);
-            TestSetup.setupScreenSize(app);
+            setupScreenSize(app);

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "setupScreenSize()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS2259)